proposals/list-patterns.md

```diff
@@ -0,0 +1,129 @@
+# List patterns
+
+## Summary
+
+Lets you to match an array or a list with a sequence of patterns e.g. `array is {1, 2, 3}` will match an integer array of the length three with 1, 2, 3 as its elements, respectively.
+
+## Detailed Design
+
+The pattern syntax is modified as follow:
+
+```antlr
+recursive_pattern
```

Comment:
Previous patterns specs did away with this single syntax form. There is no `recursive_pattern` production.
See https://github.com/dotnet/csharplang/blob/a3d67d141501101af9dbb9e8192ca8b4c717e316/proposals/csharp-8.0/patterns.md#patterns

Reply:
Sorry I missed your comment last time. I was wrongly reflecting the AST so the grammar was all wrong. =(